From: Christoph Hellwig <hch@lst.de>
Subject: squashfs: don't include buffer_head.h
Date: Wed, 17 May 2023 09:16:22 +0200

Squashfs has stopped using buffers heads in 93e72b3c612adcaca1
("squashfs: migrate from ll_rw_block usage to BIO").

Link: https://lkml.kernel.org/r/20230517071622.245151-1-hch@lst.de
Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Pankaj Raghav <p.raghav@samsung.com>
Reviewed-by: Phillip Lougher <phillip@squashfs.org.uk>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/squashfs/block.c                     |    1 -
 fs/squashfs/decompressor.c              |    1 -
 fs/squashfs/decompressor_multi_percpu.c |    1 -
 3 files changed, 3 deletions(-)

--- a/fs/squashfs/block.c~squashfs-dont-include-buffer_headh
+++ a/fs/squashfs/block.c
@@ -18,7 +18,6 @@
 #include <linux/vfs.h>
 #include <linux/slab.h>
 #include <linux/string.h>
-#include <linux/buffer_head.h>
 #include <linux/bio.h>
 
 #include "squashfs_fs.h"
--- a/fs/squashfs/decompressor.c~squashfs-dont-include-buffer_headh
+++ a/fs/squashfs/decompressor.c
@@ -11,7 +11,6 @@
 #include <linux/types.h>
 #include <linux/mutex.h>
 #include <linux/slab.h>
-#include <linux/buffer_head.h>
 
 #include "squashfs_fs.h"
 #include "squashfs_fs_sb.h"
--- a/fs/squashfs/decompressor_multi_percpu.c~squashfs-dont-include-buffer_headh
+++ a/fs/squashfs/decompressor_multi_percpu.c
@@ -7,7 +7,6 @@
 #include <linux/types.h>
 #include <linux/slab.h>
 #include <linux/percpu.h>
-#include <linux/buffer_head.h>
 #include <linux/local_lock.h>
 
 #include "squashfs_fs.h"
_
